projects/batfish/src/main/java/org/batfish/main/Batfish.java

```diff
@@ -2120,78 +2120,108 @@ public void initBgpOriginationSpaceExplicit(Map<String, Configuration> configura
+      convertAnswer
+          .getWarnings()
+          .forEach(
+              (hostname, convertWarnings) -> {
+                org.batfish.common.Warnings combined = warnings.get(hostname);
+                if (combined == null) {
+                  warnings.put(hostname, convertWarnings);
+                } else {
+                  combined.getPedanticWarnings().addAll(convertWarnings.getPedanticWarnings());
+                  combined.getRedFlagWarnings().addAll(convertWarnings.getRedFlagWarnings());
+                  combined
+                      .getUnimplementedWarnings()
+                      .addAll(convertWarnings.getUnimplementedWarnings());
+                }
+              });
+    }
+    answerElement.setParseStatus(parseAnswer.getParseStatus());
+    answerElement.setParseTrees(parseAnswer.getParseTrees());
+    for (String failed : convertAnswer.getFailed()) {
+      answerElement.getParseStatus().put(failed, ParseStatus.FAILED);
+    }
+    _logger.info(answerElement.prettyPrint());
+    return answerElement;
+  }
+
+  public InitInfoAnswerElement initInfoBgpAdvertisements(boolean summary, boolean verboseError) {
+    InitInfoAnswerElement answerElement = new InitInfoAnswerElement();
+    ParseEnvironmentBgpTablesAnswerElement parseAnswer =
+        loadParseEnvironmentBgpTablesAnswerElement();
+    if (!summary) {
+      if (verboseError) {
+        SortedMap<String, Set<BatfishStackTrace>> errors = answerElement.getErrors();
+        parseAnswer
+            .getErrors()
+            .forEach(
+                (hostname, parseErrors) -> {
+                  errors.computeIfAbsent(hostname, k -> new HashSet<>()).add(parseErrors);
+                });
+        parseAnswer
+            .getErrors()
+            .forEach(
+                (hostname, convertErrors) -> {
+                  errors.computeIfAbsent(hostname, k -> new HashSet<>()).add(convertErrors);
+                });
+      }
+      SortedMap<String, org.batfish.common.Warnings> warnings = answerElement.getWarnings();
+      warnings.putAll(parseAnswer.getWarnings());
+    }
+    answerElement.setParseStatus(parseAnswer.getParseStatus());
+    answerElement.setParseTrees(parseAnswer.getParseTrees());
+    _logger.info(answerElement.prettyPrint());
+    return answerElement;
+  }
+
+  public InitInfoAnswerElement initInfoRoutes(boolean summary, boolean verboseError) {
+    InitInfoAnswerElement answerElement = new InitInfoAnswerElement();
+    ParseEnvironmentRoutingTablesAnswerElement parseAnswer =
+        loadParseEnvironmentRoutingTablesAnswerElement();
+    if (!summary) {
+      if (verboseError) {
+        SortedMap<String, Set<BatfishStackTrace>> errors = answerElement.getErrors();
+        parseAnswer
+            .getErrors()
+            .forEach(
+                (hostname, parseErrors) -> {
+                  errors.computeIfAbsent(hostname, k -> new HashSet<>()).add(parseErrors);
+                });
+        parseAnswer
```

Comment:
Seems like this pattern of copying errors from an answerElement to a map / warnings from an answerElement to a map has opportunity for some code reuse.